### ISSUE TITLE
Uncaught TypeError: hook.error is not a function MochaAllureReporter

### ISSUE THREAD

**Issue by darkallure** [allure-framework/allure-js#482]
**Describe the bug**
Getting an error when parallel mode is on while using Mocha

**To Reproduce**
1. Set parallel mode to true in mocha.rc config or use "--parallel" flag
2. Run the tests
3. Tests fail due to the following error: Uncaught TypeError: hook.error is not a function

**Expected behavior**
The tests should run successfully in parallel mode with Mocha allure reporter
**Screenshots**
![image](https://user-images.githubusercontent.com/20166110/185558686-3f8b72ce-1982-4e3f-9630-845826a39fec.png)
![image](https://user-images.githubusercontent.com/20166110/185558794-c7968a0b-f4f1-4bf8-95ff-acbcf3e11c38.png)

**Additional context**
the following version packages are used:
    "allure-mocha": "^2.0.0-beta.19",
    "mocha": "^9.2.1"

**Comment by jamesmortensen** [comment]
@darkallure if you patch MochaAllureReporter on line 70 with this block of code, do you still get the error?  This resolved the issue for me, and I've opened a pull request to get the changes reviewed and merged:

**node_modules/allure-mocha/dist/MochaAllureReporter.js line 70:**
```javascript
 onHookEnd(hook) {
        console.log('inside onHookEnd...');
        if(typeof hook.error === 'function')
            this.coreReporter.endHook(hook.error());
        else
            this.coreReporter.endHook();
    }
```

**Comment by darkallure** [comment]
@jamesmortensen  thank you so much! It did fixed the issue but the other issue started to occur, do you know how to fix this?
![image](https://user-images.githubusercontent.com/20166110/185745764-36afb2db-2a69-4264-9318-305beff0d864.png)

**Comment by jamesmortensen** [comment]
@darkallure Not sure if that's related to this change or if it's another issue altogether which needs to be reported.  It does seem like the v2.0.0 platform is still missing many of the 1.0 features, so my guess is new issue.  If you check the source code, do you see any implementation for label?

**Comment by darkallure** [comment]
Yes, I do see it: 
![image](https://user-images.githubusercontent.com/20166110/185859695-aa7ddf2a-695e-4c06-971c-5060a14ab201.png)
The issue is not only with label but all the other properties as well: allure.suite, allure.description etc.

**Comment by jamesmortensen** [comment]
Can you try using the JavaScript Debug Terminal in VSCode to step through code in order to see what's missing?  My guess is something is going on in the "model" that's being required.

**Comment by darkallure** [comment]
@jamesmortensen  it cannot even pass this line: 
![image](https://user-images.githubusercontent.com/20166110/186507763-6c22c2a2-811e-41b4-948c-51b8cc93d414.png)
Do you have any other suggestions on what can be done to fix it or some kind of a workaround?

**Comment by jamesmortensen** [comment]
I'm not yet sure what runtime is supposed to be or where it comes from.  It's supposed to be passed into the Allure constructor.  I'm thinking it must be related to Mocha or Jest or Jasmine or whatever the runner is that's being used, but I'm not sure.  Hopefully one of the maintainers of the project can jump in and give some insight.

**Comment by delatrie** [comment]
Fixed in 2.0.0.beta-20 via #510. Remains fixed in 3.0.0-beta.3, which uses a different approach to the parallel mode.
